aea/test_tools/test_cases.py

```diff
@@ -182,6 +184,29 @@ def _run_python_subprocess(self, *args: str) -> subprocess.Popen:
         self.subprocesses.append(process)
         return process
 
+    def is_present_in_output(
+        self,
+        process: subprocess.Popen,
+        string: str,
+        timeout: int = DEFAULT_PROCESS_TIMEOUT,
+    ) -> bool:
+        """
+        Check if string present in agent process output.
+
+        :param process: a subprocess object to communicate.
+        :param string: a substring expected to appear in process output.
+        :param timeout: a timeout in seconds.
+
+        :return: boolean is string in process output.
+        """
+        try:
+            output, _err = process.communicate(timeout=timeout)
+        except subprocess.TimeoutExpired:
+            self.terminate_agents(process)
+            output, _err = process.communicate()
+
+        return string in str(output)
```

Comment:
I think you'll need a loop here. The idea in pseudocode would be:
while time < timeout:
    # check if strings already present in output.
    if not:
        sleep for 0.5 sec
    else:
        break

Reply:
The current implementation means you will always have to wait until the end of the timeout. But we want to succeed as early as we can to save time.

Reply:
Tried that, not gonna work. `process.communicate()` stuck if trying to communicate to an active process.